File: Data Structures & Algorithms/merge-intervals/submission-1.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        n = len(intervals)
        intervals.sort()
        res = []

        for i in range(n):
            interval = intervals[i]
            while res and res[-1][1] >= interval[0]:
                interval = [min(res[-1][0], interval[0]), max(res[-1][1], interval[1])]
                res.pop()
            res.append(interval)
        return res                 

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = prices[0];
        int profit = 0;

        for (int x : prices) {
            minPrice = min(minPrice, x);
            profit = max(profit, x-minPrice);
        }
        return profit;
    }
};
